### PR TITLE
Scan the shared external storage source on access

### LINKED ISSUE #23988: Files uploaded to external storage not shown immediately in Webinterface in shared folders
Hello
There seems to be a problem in NC20.01 and external storages. When I place new files on external storage, they are not shown in WebInterface as normal. I get this Problem after upgrading from 19.04 to NC20.01.

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix #23988
Regression from #22867

### CODE REVIEW

**Review by icewind1991 — DISMISSED** [review]
This should instead be done by changing the `Watcher` returned by `SharedStorage::getWatcher`.

It currently returns a `NullWatcher` since it assumes that the share source handles it's updating (which is true for non-external-storage shares).

**Comment by skjnldsv** [comment]
@icewind1991 I knew there was a better way :sob: 
Shouldn't we return depending if the SharedStorage is an external only?
Or should we always return another watcher, I guess it's a regression from https://github.com/nextcloud/server/pull/22867/

**Comment by PVince81** [comment]
@CarlSchwan can you check if this might be related to the issue you're looking into ? seems it's also about a mount not being considered properly.

**Review by PVince81 — APPROVED** [review]
:+1: looks fine to me, but would be good to get confirmation from @icewind1991 also

**Review by icewind1991 — DISMISSED** [review]

File: apps/files_sharing/lib/SharedStorage.php [hunk 1]
```diff
@@ -405,7 +407,11 @@ public function getOwner($path): string {
 		return $this->superShare->getShareOwner();
 	}
 
-	public function getWatcher($path = '', $storage = null): NullWatcher {
+	public function getWatcher($path = '', $storage = null): Watcher {
+		// If the share is originating from an external storage
+		if ($this->getShare()->getNode()->getMountPoint() instanceof ExternalMountPoint) {
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply apps/files_sharing/lib/SharedStorage.php]: Done, please validate the new approach :)
No idea why `$mountManager->findByNumericId` is returning an array though :thinking:

**Review by CarlSchwan — APPROVED** [review]
Lgtm 🙂

**Comment by skjnldsv** [comment]
Failures unrelated

**Pull request merged**